source/io.c

```diff
@@ -609,3 +610,143 @@ PyObject *aws_py_tls_connection_options_set_server_name(PyObject *self, PyObject
 
     Py_RETURN_NONE;
 }
+
+/* aws_input_stream implementation for accessing Python I/O classes */
+struct aws_input_stream_py_impl {
+    struct aws_input_stream base;
+
+    bool is_end_of_stream;
+
+    /* Dependencies that must outlive this */
+    PyObject *io;
+};
+
+static void s_aws_input_stream_py_clean_up(struct aws_input_stream *stream) {
+    struct aws_input_stream_py_impl *impl = stream->impl;
+    Py_DECREF(impl->io);
+    aws_mem_release(aws_py_get_allocator(), impl);
+}
+
+static int s_aws_input_stream_py_seek(
+    struct aws_input_stream *stream,
+    aws_off_t offset,
+    enum aws_stream_seek_basis basis) {
+
+    struct aws_input_stream_py_impl *impl = stream->impl;
+
+    int aws_result = AWS_OP_SUCCESS;
+    PyObject *method_result = NULL;
+
+    /*************** GIL ACQUIRE ***************/
+    PyGILState_STATE state = PyGILState_Ensure();
+
+    method_result = PyObject_CallMethod(impl->io, "seek", "(li)", &offset, &basis);
+    if (!method_result) {
+        aws_result = aws_py_raise_error();
+        goto done;
+    }
+
+    /* Clear EOF */
+    impl->is_end_of_stream = false;
```

Comment:
What if someone seeks to END, 0? Should we check?

Reply:
it would get set again the next time they tried to read.
This behavior seems consistent with [std::iostream](https://en.cppreference.com/w/cpp/io/ios_base/iostate)
> The following functions clear eofbit as a side-effect:
> -  basic_istream::putback
> -  basic_istream::unget
> -  basic_istream::seekg

Reply:
:+1: